tests/asgi/tests.py

```diff
@@ -254,7 +279,7 @@ async def test_assert_in_listen_for_disconnect(self):
         await communicator.send_input({"type": "http.not_a_real_message"})
         msg = "Invalid ASGI message after request body: http.not_a_real_message"
         with self.assertRaisesMessage(AssertionError, msg):
-            await communicator.receive_output()
+            await communicator.wait()
```

Comment:
This causes `ResourceWarning` on Python 3.12+:
```
./runtests.py asgi.tests.ASGITest.test_assert_in_listen_for_disconnect --parallel=1
Testing against Django installed in '/django/django'
Found 1 test(s).
System check identified no issues (0 silenced).
.
----------------------------------------------------------------------
Ran 1 test in 0.004s

OK
Exception ignored in: <function SpooledTemporaryFile.__del__ at 0x7ff588dc09a0>
Traceback (most recent call last):
  File "/usr/local/lib/python3.12/tempfile.py", line 736, in __del__
    _warnings.warn(
ResourceWarning: Unclosed file <tempfile.SpooledTemporaryFile object at 0x7ff586f9da50>
```
Any ideas?

Reply:
I added a possible fix. Can anyone take a look?

Reply:
@felixxm The fix in  08e72a6413bbd66e4d39a89b5b9c31c38408289f looks correct: it's restoring the behaviour from before this change. (Note the same `except` block in the previous version.) The test exercises the `assert` for the (protocol server programming error) invalid message type in `listen_for_disconnect()`. 

It would have been dropped here because @qxcv proposed moving to use a context manager block for the body file here. See https://github.com/django/django/pull/17147#pullrequestreview-1564929017

👍

Reply:
Thanks for checking :+1: 

> That may be worth it, if you think, as an improvement — but a part of me thinks there **really should** only be a small finite set of ways out of this block and I'd rather blow up if we've missed one... 🤔

Yeah, we can leave it for a separate discussion/PR.